### PR TITLE
RF: Moved Analytics to class

### PR DESCRIPTION
**Pull request by blckmn**
Encapsulating the Analytics in a class.

### CODE REVIEW

File: src/js/serial_backend.js
Comment (chmelevskij) [inline src/js/serial_backend.js:160]: I think this won't work. I can't see `tracking` imported or bound to `window` anywhere.

**Review by chmelevskij — COMMENTED** [review]
I'm also getting following error on master
```
object_hash.js:1 Uncaught Error: Object argument required.
    at u (object_hash.js:1:2030)
    at t (object_hash.js:1:753)
    at Function.sha1 (object_hash.js:1:1085)
    at serial_backend.js:443:77
    at Object.send_message (msp.js:323:17)
    at processUid (serial_backend.js:440:9)
    at serial_backend.js:435:9
    at Object.send_message (msp.js:323:17)
    at checkReportProblems (serial_backend.js:403:9)
    at processBoardInfo (serial_backend.js:383:9)
    at onOpenVirtual (serial_backend.js:325:5)
    at Object.connectVirtual (serial.js:215:13)
    at Object.connect (serial.js:33:18)
    at HTMLAnchorElement.<anonymous> (serial_backend.js:75:32)
    at HTMLAnchorElement.dispatch (jquery.min.js:2:43064)
    at y.handle (jquery.min.js:2:41048)
u @ object_hash.js:1
t @ object_hash.js:1
sha1 @ object_hash.js:1
(anonymous) @ serial_backend.js:443
send_message @ msp.js:323
processUid @ serial_backend.js:440
(anonymous) @ serial_backend.js:435
send_message @ msp.js:323
checkReportProblems @ serial_backend.js:403
processBoardInfo @ serial_backend.js:383
onOpenVirtual @ serial_backend.js:325
connectVirtual @ serial.js:215
connect @ serial.js:33
(anonymous) @ serial_backend.js:75
dispatch @ jquery.min.js:2
y.handle @ jquery.min.js:2
```

**Comment by limonspb** [comment]
> I'm also getting following error on master
> 
> ```
> object_hash.js:1 Uncaught Error: Object argument required.
>     at u (object_hash.js:1:2030)
>     at t (object_hash.js:1:753)
>     at Function.sha1 (object_hash.js:1:1085)
>     at serial_backend.js:443:77
>     at Object.send_message (msp.js:323:17)
>     at processUid (serial_backend.js:440:9)
>     at serial_backend.js:435:9
>     at Object.send_message (msp.js:323:17)
>     at checkReportProblems (serial_backend.js:403:9)
>     at processBoardInfo (serial_backend.js:383:9)
>     at onOpenVirtual (serial_backend.js:325:5)
>     at Object.connectVirtual (serial.js:215:13)
>     at Object.connect (serial.js:33:18)
>     at HTMLAnchorElement.<anonymous> (serial_backend.js:75:32)
>     at HTMLAnchorElement.dispatch (jquery.min.js:2:43064)
>     at y.handle (jquery.min.js:2:41048)
> u @ object_hash.js:1
> t @ object_hash.js:1
> sha1 @ object_hash.js:1
> (anonymous) @ serial_backend.js:443
> send_message @ msp.js:323
> processUid @ serial_backend.js:440
> (anonymous) @ serial_backend.js:435
> send_message @ msp.js:323
> checkReportProblems @ serial_backend.js:403
> processBoardInfo @ serial_backend.js:383
> onOpenVirtual @ serial_backend.js:325
> connectVirtual @ serial.js:215
> connect @ serial.js:33
> (anonymous) @ serial_backend.js:75
> dispatch @ jquery.min.js:2
> y.handle @ jquery.min.js:2
> ```
> 
> At least in virtual mode. And assume this exists in this pr. Could we fix that while we are at it?

Is that on Receiver tab?
Sorry not on PC now.

**Comment by limonspb** [comment]
> I'm also getting following error on master
> 
> ```
> object_hash.js:1 Uncaught Error: Object argument required.
>     at u (object_hash.js:1:2030)
>     at t (object_hash.js:1:753)
>     at Function.sha1 (object_hash.js:1:1085)
>     at serial_backend.js:443:77
>     at Object.send_message (msp.js:323:17)
>     at processUid (serial_backend.js:440:9)
>     at serial_backend.js:435:9
>     at Object.send_message (msp.js:323:17)
>     at checkReportProblems (serial_backend.js:403:9)
>     at processBoardInfo (serial_backend.js:383:9)
>     at onOpenVirtual (serial_backend.js:325:5)
>     at Object.connectVirtual (serial.js:215:13)
>     at Object.connect (serial.js:33:18)
>     at HTMLAnchorElement.<anonymous> (serial_backend.js:75:32)
>     at HTMLAnchorElement.dispatch (jquery.min.js:2:43064)
>     at y.handle (jquery.min.js:2:41048)
> u @ object_hash.js:1
> t @ object_hash.js:1
> sha1 @ object_hash.js:1
> (anonymous) @ serial_backend.js:443
> send_message @ msp.js:323
> processUid @ serial_backend.js:440
> (anonymous) @ serial_backend.js:435
> send_message @ msp.js:323
> checkReportProblems @ serial_backend.js:403
> processBoardInfo @ serial_backend.js:383
> onOpenVirtual @ serial_backend.js:325
> connectVirtual @ serial.js:215
> connect @ serial.js:33
> (anonymous) @ serial_backend.js:75
> dispatch @ jquery.min.js:2
> y.handle @ jquery.min.js:2
> ```
> 
> At least in virtual mode. And assume this exists in this pr. Could we fix that while we are at it?

Found it!
fixing here:
https://github.com/betaflight/betaflight-configurator/pull/3223

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> FAIL
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> FAIL
 - assigned to an approver -> PASS
 - approver count at least three -> FAIL

File: src/js/serial_backend.js
Comment (blckmn) [reply src/js/serial_backend.js:160]: it is set on line 122 in main.js

**Review by chmelevskij — APPROVED** [review]

**Review by haslinghuis — APPROVED** [review]

**Pull request merged**
